V/VMEC/build_tarballs.jl

```diff
@@ -1,41 +1,73 @@
 ]
 
 script = raw"""
-mkdir -p ${libdir}
-cd ${WORKSPACE}/srcdir/VMEC2000*
-make USE_MKL=1
-install -m755 lib/libvmec_mkl.so ${libdir}
-make USE_MKL=1 vmec_clean ; make USE_MKL=0
-install -m755 lib/libvmec_openblas.so ${libdir}
-make USE_MKL=0 vmec_clean
+cd ${WORKSPACE}/srcdir/VMEC*
+if [[ ${target} == *mingw* ]]; then
+  sed "s/LT_INIT/LT_INIT(win32-dll)/" configure.ac | cat > configure.ac.2
+  mv configure.ac.2 configure.ac
+  ./autogen.sh
+  ./configure CC=gcc FC=gfortran F77=gfortran --build=${MACHTYPE} --with-mkl --host=${target} --target=${target} --prefix=${prefix}
+  # Deal with the issue that gcc doesn't accept -no-undefined at configure step
+  sed "s/AM_LDFLAGS =/AM_LDFLAGS =-no-undefined/" Makefile | cat > Makefile.2
+  mv Makefile.2 Makefile
+  make && make install && make clean
+
+  ./configure CC=gcc FC=gfortran F77=gfortran --build=${MACHTYPE} --host=${target} --target=${target} --prefix=${prefix}
+  # Deal with the issue that gcc doesn't accept -no-undefined at configure step
+  sed "s/AM_LDFLAGS =/AM_LDFLAGS =-no-undefined/" Makefile | cat > Makefile.2
+  mv Makefile.2 Makefile
+  make && make install && make clean
+
+else
+  # Configure with MKL provided ScaLAPACK
+  ./autogen.sh
+  ./configure CC=mpicc FC=mpifort F77=mpifort FFLAGS=-O3 FCFLAGS=-O3 --with-mkl --build=${MACHTYPE} --host=${target} --target=${target} --prefix=${prefix}
+  make && make install && make clean
+
+  # Configure with OpenBLAS and ScaLAPACK from SCALAPACK_jll
+  ./configure CC=mpicc FC=mpifort F77=mpifort FFLAGS=-O3 FCFLAGS=-O3 --build=${MACHTYPE}  --host=${target} --target=${target} --prefix=${prefix}
+  make && make install && make clean
+fi
 """
 
-platforms = [
-    Platform("x86_64", "linux"; libc = "glibc", libgfortran_version="4.0.0"),
-    Platform("x86_64", "linux"; libc = "glibc", libgfortran_version="5.0.0"),
-]
-#platforms = expand_gfortran_versions(platforms)
+incompatible_arch = ["i686", "armv6l", "armv7l", "aarch64", "powerpc64le"]
+incompatible_os = ["freebsd"]
+
+platforms = expand_gfortran_versions(supported_platforms())
+
+# Filter out libgfortran_version = 3.0.0 which is incompatible with VMEC
+filter!(p ->libgfortran_version(p) >= v"4", platforms)
+
+# Filter incompatible architectures and operating systems
+filter!(p -> arch(p) == "x86_64", platforms)
+filter!(!Sys.isfreebsd, platforms)
+
+# Right now VMEC only works with libc=glibc, filter out any musl dependencies
+filter!(p -> !Sys.islinux(p) || libc(p) == "glibc", platforms)
 
 # The products that we will ensure are always built
+# Don't automatically dl_open so that the appropriate 
+# library can be loaded on intiation of VMEC.jl
 products = [
     LibraryProduct("libvmec_mkl", :libvmec_mkl, dont_dlopen = true),
     LibraryProduct("libvmec_openblas", :libvmec_openblas, dont_dlopen = true),
 ]
 
 # Dependencies that must be installed before this package can be built
 dependencies = [
+    Dependency("MbedTLS_jll", v"2.24.0"),
```

Comment:
Ok, now I see what's the problem with MbedTLS (although I still don't understand why the wrong version is installed).

However I'd make it a `BuildDependency`, this package doesn't strictly need it
```suggestion
    # MbedTLS is an indirect dependency, fix the version for building 
    BuildDependency(PackageSpec("MbedTLS_jll", v"2.24.0")),
```
Note that you'll need to load also `Pkg` above

Reply:
Makes sense, I'm slowly understanding all the details involved.